### ISSUE TITLE
Export-M365DSCConfiguration App permissions needed ApplicationId

### ISSUE THREAD

**Issue by Phentrin** [microsoft/Microsoft365DSC#2149]
I'm trying to export the configuration with ApplicationId/secret parameters.
input: https://microsoft365dsc.com/user-guide/get-started/snapshot-of-existing-tenant/

**Module Version:**
Manifest   1.22.720.1 Microsoft365DSC       

**Commande Used**
Export-M365DSCConfiguration -Components @("SPOAccessControlSettings", "SPOApp", "SPOBrowserIdleSignout", "SPOHomeSite", "SPOHubSite", "SPOOrgAssetsLibrary", "SPOPropertyBag", "SPOSearchManagedProperty", "SPOSearchResultSource", "SPOSharingSettings", "SPOSite", "SPOSiteAuditSettings", "SPOSiteDesign", "SPOSiteDesignRights", "SPOSiteGroup", "SPOSiteScript", "SPOStorageEntity", "SPOTenantCdnEnabled", "SPOTenantCdnPolicy", "SPOTenantSettings", "SPOTheme", "SPOUserProfileProperty") -ApplicationId $ApplicationId -ApplicationSecret $ApplicationSecret -TenantId $TenantId -Path "D:\Temp\DSCExtracts\" -ConfigurationName MyConfig -FileName MyExport.ps1

I added/approved all the API Graph Permissions needed (especially for SPO)
![image](https://user-images.githubusercontent.com/12797998/181499239-be249f9e-6271-4ce4-9645-16bc95fb3108.png)

But it seems some of the resources cannot be exported. 
![image](https://user-images.githubusercontent.com/12797998/181498656-d22f2e62-643c-41aa-96cc-6c765eb09f72.png)

Any recommandation how to resolve that ?

**Comment by andikrueger** [comment]
Please have a look at this part of the documentation on how to generate a list of required permissions:

https://microsoft365dsc.com/user-guide/get-started/authentication-and-permissions/#determine-required-permissions

PLase make sure to have the following permissions set:

``` powershell
> $permissions = Get-M365DSCCompiledPermissionList -ResourceNameList @("SPOAccessControlSettings", "SPOApp", "SPOBrowserIdleSignout", "SPOHomeSite", "SPOHubSite", "SPOOrgAssetsLibrary", "SPOPropertyBag", "SPOSearchManagedProperty", "SPOSearchResultSource", "SPOSharingSettings", "SPOSite", "SPOSiteAuditSettings", "SPOSiteDesign", "SPOSiteDesignRights", "SPOSiteGroup", "SPOSiteScript", "SPOStorageEntity", "SPOTenantCdnEnabled", "SPOTenantCdnPolicy", "SPOTenantSettings", "SPOTheme", "SPOUserProfileProperty")
> $permissions.ReadPermissions
Domain.Read.All
Sites.FullControl.All
DeviceManagementServiceConfig.Read.All
Group.Read.All
Application.Read.All

**Comment by philippe-entringer** [comment]
Yes I give permissions to the 
$permissions = Get-M365DSCCompiledPermissionList -ResourceNameList @("SPOAccessControlSettings", "SPOApp", "SPOBrowserIdleSignout", "SPOHomeSite", "SPOHubSite", "SPOOrgAssetsLibrary", "SPOPropertyBag", "SPOSearchManagedProperty", "SPOSearchResultSource", "SPOSharingSettings", "SPOSite", "SPOSiteAuditSettings", "SPOSiteDesign", "SPOSiteDesignRights", "SPOSiteGroup", "SPOSiteScript", "SPOStorageEntity", "SPOTenantCdnEnabled", "SPOTenantCdnPolicy", "SPOTenantSettings", "SPOTheme", "SPOUserProfileProperty")
$permissions.ReadPermissions

**Domain.Read.All
Sites.FullControl.All
DeviceManagementServiceConfig.Read.All
Group.Read.All
Application.Read.All**

And Also give permissions to the "Microsoft Graph PowerShell"
Update-M365DSCAllowedGraphScopes -All -Type Read

But I still have this result
Connecting to {PnP}...✅
[1/22] Extracting [SPOAccessControlSettings]...✅
[2/22] Extracting [SPOApp]...❌
[3/22] Extracting [SPOBrowserIdleSignout]...✅
[4/22] Extracting [SPOHomeSite]...✅
[5/22] Extracting [SPOHubSite]...❌
[6/22] Extracting [SPOOrgAssetsLibrary]...❌
[7/22] Extracting [SPOPropertyBag]...❌
[8/22] Extracting [SPOSearchManagedProperty]...❌
[9/22] Extracting [SPOSearchResultSource]...❌
[10/22] Extracting [SPOSharingSettings]...

Any idea ?

**Comment by andikrueger** [comment]
Please make sure you did register the pnp powershell within the tenant. This is a separate step.

**Comment by NikCharlebois** [comment]
Very strange indeed. I am able to replicate. I works fine if you use CertificateThumbprint instead of App Secret, so it is not a permission scope issue. Will investigate further.

**Comment by NikCharlebois** [comment]
It appears that the legacy Authentication isn't properly working with various scopes:
https://pnp.github.io/powershell/cmdlets/Connect-PnPOnline.html#sharepoint-acs-legacy-app-only

@philippe-entringer recommendation until we make a decision on how to handle this is to use Certificate Thumbprint instead of App Secret, which will make the whole process more secure anyway. Thanks

// CC @desmay we will need to decide whether or not we continue to support AppSecret for the ODSP workload.

**Comment by philippe-entringer** [comment]
Thanks Nik, 
I will use Thumbprint for the Authentication.

**Comment by OlivierWaterloos** [comment]
I have the same issue with Certificate Thumprint:
Export-M365DSCConfiguration -Path $DSCPathWork -FileName $FilePSName  -Workloads @("$WorkLoad") -ApplicationId $ApplicationId -CertificateThumbprint $CertificateThumbprint -TenantId $TenantId

**Comment by andikrueger** [comment]
Could you verify if this is still an issue with the latest version of M365DSC?

**Comment by Chabango** [comment]
I am seeing this same issue with the SharePoint module using certificate thumbprint.
However, even though I have added the Sites.FullControl.All to the app, it is not showing when I get the read permissions:

`$permissions = Get-M365DSCCompiledPermissionList -ResourceNameList @("SPOAccessControlSettings", "SPOApp", "SPOBrowserIdleSignout", "SPOHomeSite", "SPOHubSite", "SPOOrgAssetsLibrary", "SPOPropertyBag", "SPOSearchManagedProperty", "SPOSearchResultSource", "SPOSharingSettings", "SPOSite", "SPOSiteAuditSettings", "SPOSiteDesign", "SPOSiteDesignRights", "SPOSiteGroup", "SPOSiteScript", "SPOStorageEntity", "SPOTenantCdnEnabled", "SPOTenantCdnPolicy", "SPOTenantSettings", "SPOTheme", "SPOUserProfileProperty")
$permissions.ReadPermissions`

PS C:\Users\XYZ> $permissions = Get-M365DSCCompiledPermissionList -ResourceNameList @("SPOAccessControlSettings", "SPOApp", "SPOBrowserIdleSignout", "SPOHomeSite", "SPOHubSite", "SPOOrgAssetsLibrary", "SPOPropertyBag", "SPOSearchManagedProperty", "SPOSearchResultSource", "SPOSharingSettings", "SPOSite", "SPOSiteAuditSettings", "SPOSiteDesign", "SPOSiteDesignRights", "SPOSiteGroup", "SPOSiteScript", "SPOStorageEntity", "SPOTenantCdnEnabled", "SPOTenantCdnPolicy", "SPOTenantSettings", "SPOTheme", "SPOUserProfileProperty")
$permissions.ReadPermissions

Domain.Read.All
DeviceManagementServiceConfig.Read.All
Group.Read.All
Application.Read.All

Any suggestions on how to make sure the Sites.FullControl.All is applied. It has been almost 48 hours since I applied it.
